lib/recipe.rb

```diff
@@ -145,11 +148,19 @@
   %i{windows}  => nil
 )
 
-# The windows_user resource isn't idempotent
+use_special_macos_user = platform?("mac_os_x") && (Gem::Version.new(node["platform_version"]) >= Gem::Version.new("10.14"))
+
+liveness_agent_macos_user agent_username do
+  home agent_dir
+  shell agent_user_shell
+  only_if { use_special_macos_user }
```

Comment:
I wonder if it might be better to do this as an if block around the entire resource.  I would have expected that even with this only_if, things like `load_current_resource` might still get called.  Or rather, I'm not sure there is a guarantee it won't get called.

Reply:
`Chef::Resource` evaluates guards and determines if it should skip before it loads the provider for the resource and call `load_current_resource`

Reply:
https://github.com/chef/chef/blob/master/lib/chef/resource.rb#L584

Reply:
I'd probably still go with an outer if statement to avoid any confusing logging and to make sure our recipe isn't depending on behavior that isn't specified (although I'm sure it is very unlikely to change).  Either way, marking as "resolved", thanks for the code pointer!

Reply:
I changed it for ya anyway :hamburger: